addons/web/static/src/js/core/owl_popover.js

```diff
@@ -0,0 +1,272 @@
+     **/
+    class Popover extends Component {
+        /**
+         * @param {Object} [props]
+         * @param {string} [props.position='bottom'] 'top', 'bottom', 'left' or 'right'
+         * @param {HTMLElement} [props.target='body'] the target element (the popover's
+         *                      arrow decoration will point to it)
+         * @param {string} [props.title]
+         */
+        constructor() {
+            super(...arguments);
+            this.popoverRef = useRef('popover');
+            this.orderedPositions = ['top', 'right', 'bottom', 'left'];
+            this.requestedPosition = null;
+            this.state = useState({
+                displayed: false,
+            })
+            // without event capture, scroll listener on children nodes will prevent
+            // this handler from being fired
+            useExternalListener(document, 'scroll', this.show, {
+                capture: true,
+            });
+            useExternalListener(window, 'resize', this.show);
+        }
+
+        patched() {
+            super.patched(...arguments);
+            this.show();
+        }
+
+        /**
+         * Show the popover according to its props. This method will try to position the
+         * popover as request (according to the `position` props) but will fallback
+         * on 'bottom' positioning if the requested position would put it outside of the
+         * viewport.
+         */
+        show() {
+            if (!this.state.displayed) {
+                return;
+            }
+            // get the target from the dom
+            // we don't want to do this early, since the target might not
+            // be in the dom when the component is instanciated
+            if (!this.constructor._isInViewport(this.el)) {
+                // target is no longer in the viewport, close the popover
+                return this._close();
+            }
+            this.requestedPosition = this.props.position;
+
+            const positionIndex = this.orderedPositions.indexOf(
+                this.requestedPosition
+            );
+
+            const positioningData = this.constructor._computePositioningData(
+                this.popoverRef.el,
+                this.el
+            );
+
+            // check if the requested position fits the viewport; if not,
+            // try all other positions and find one that does
+            const position = this.orderedPositions
+                .slice(positionIndex)
+                .concat(this.orderedPositions.slice(0, positionIndex))
+                .map((pos) => positioningData[pos])
+                .find((pos) => {
+                    this.popoverRef.el.style.top = `${pos.top}px`;
+                    this.popoverRef.el.style.left = `${pos.left}px`;
+                    return this.constructor._isInViewport(this.popoverRef.el);
+                });
+
+            // remove all positioning classes
+            this.orderedPositions.forEach((pos) => {
```

Comment:
shouldn't you do that before trying to find the correct position? CSS rules may affect the position...

Reply:
:+1: